Documentation/ApiOverview/Assets/Index.rst

```diff
@@ -0,0 +1,124 @@
+- :php:`\TYPO3\CMS\Core\Page\AssetCollector::addInlineJavaScript(string $identifier, string $source, array $attributes, array $options = []): self`
+- :php:`\TYPO3\CMS\Core\Page\AssetCollector::addStyleSheet(string $identifier, string $source, array $attributes, array $options = []): self`
+- :php:`\TYPO3\CMS\Core\Page\AssetCollector::addInlineStyleSheet(string $identifier, string $source, array $attributes, array $options = []): self`
+- :php:`\TYPO3\CMS\Core\Page\AssetCollector::addMedia(string $fileName, array $additionalInformation): self`
+- :php:`\TYPO3\CMS\Core\Page\AssetCollector::removeJavaScript(string $identifier): self`
+- :php:`\TYPO3\CMS\Core\Page\AssetCollector::removeInlineJavaScript(string $identifier): self`
+- :php:`\TYPO3\CMS\Core\Page\AssetCollector::removeStyleSheet(string $identifier): self`
+- :php:`\TYPO3\CMS\Core\Page\AssetCollector::removeInlineStyleSheet(string $identifier): self`
+- :php:`\TYPO3\CMS\Core\Page\AssetCollector::removeMedia(string $identifier): self`
+- :php:`\TYPO3\CMS\Core\Page\AssetCollector::getJavaScripts(?bool $priority = null): array`
+- :php:`\TYPO3\CMS\Core\Page\AssetCollector::getInlineJavaScripts(?bool $priority = null): array`
+- :php:`\TYPO3\CMS\Core\Page\AssetCollector::getStyleSheets(?bool $priority = null): array`
+- :php:`\TYPO3\CMS\Core\Page\AssetCollector::getInlineStyleSheets(?bool $priority = null): array`
+- :php:`\TYPO3\CMS\Core\Page\AssetCollector::getMedia(): array`
+
+New ViewHelpers
+---------------
+
+There are also two new ViewHelpers, the :ref:`f:asset.css<t3viewhelper:typo3-fluid-asset-css>` and the :ref:`f:asset.script<t3viewhelper:typo3-fluid-asset-script>` ViewHelper which use the :php:`AssetCollector` API.
+
+
+Considerations
+--------------
+
+Currently, CSS and JavaScript registered with the AssetCollector will be rendered after their
+PageRenderer counterparts. The order is:
+
+- :html:`<head>`
+- :ts:`page.includeJSLibs.forceOnTop`
+- :ts:`page.includeJSLibs`
+- :ts:`page.includeJS.forceOnTop`
+- :ts:`page.includeJS`
+- :php:`AssetCollector::addJavaScript()` with 'priority'
+- :ts:`page.jsInline`
+- :php:`AssetCollector::addInlineJavaScript()` with 'priority'
+- :html:`</head>`
+
+- :ts:`page.includeJSFooterlibs.forceOnTop`
+- :ts:`page.includeJSFooterlibs`
+- :ts:`page.includeJSFooter.forceOnTop`
+- :ts:`page.includeJSFooter`
+- :php:`AssetCollector::addJavaScript()`
+- :ts:`page.jsFooterInline`
+- :php:`AssetCollector::addInlineJavaScript()`
+
+.. note::
+
+   JavaScript registered with AssetCollector is not affected by
+   :ts:`config.moveJsFromHeaderToFooter`.
+
+Examples
+--------
+
+Add a JavaScript file to the collector with script attribute :code:`data-foo="bar"`:
+
+.. code-block:: php
+
+    GeneralUtility::makeInstance(AssetCollector::class)
+       ->addJavaScript('my_ext_foo', 'EXT:my_ext/Resources/Public/JavaScript/foo.js', ['data-foo' => 'bar']);
```

Comment:
We should also explain what happens if the same asset is registered multiple times with different attributes.

Reply:
Good point. 
It merges with the existing assets - https://github.com/TYPO3/TYPO3.CMS/blob/master/typo3/sysext/core/Classes/Page/AssetCollector.php#L77.

For more control the AssetRenderer events https://docs.typo3.org/c/typo3/cms-core/master/en-us/Changelog/10.4/Feature-90899-IntroduceAssetPreRenderingEvents.html#changelog-feature-90899-introduceassetprerenderingevents shall be used.

Reply:
Thank you both for your comments (and the links), I added an explanation of the merging as well as the description of the events and cross-links to them.